README.md

```diff
@@ -107,19 +138,114 @@ scalaVersion := "3.2.1"
 
 There are many reasons why we decide to build this library, such as:
 * allowing user to build Spark pipeline with ZIO easily.
-* making better code, pure FP, more composable, more readable Spark code.
+* making *better code*, pure FP, more composable, more readable Spark code.
 * stopping the propagation of ```implicit SparkSessions```.
 * improving some performances.
 * taking advantage of ZIO allowing our jobs to retry and to be run in parallel.
 
+
+## Design
+"What if Spark was using better functional programming and an effect system?"
+
+zio-spark is built with this main idea in mind, to rewrite the existing API in Spark using better 
+functional programming principle. You will find a corresponding type for the existing API : 
+
+
+| org.apache.spark | zio.spark        |
+|------------------|------------------|
+| sql.Dataset      | sql.Dataset      |
+| sql.SparkSession | sql.SparkSession |
+| ...              | ...              |
+
+
+
+It comes with different API, for example : 
+
+```scala
```

Comment:
I am not sure if it is interesting to show the wrapping code; Under the hood we actually call the function below. It is not zio-spark compare to spark but zio-spark wrappring the spark function.